synapse/media/url_previewer.py

```diff
@@ -623,6 +649,10 @@ async def _precache_image_url(
         if not image_url:
             return
 
+        # Don't attempt to download the URL if it is blocked.
+        if self._is_url_blocked(image_url):
```

Comment:
Instead of checking against the `image_url` at this point, do we instead want to run it against the `image_url = urljoin(media_info.uri, image_url)` that we craft below (the full URL)?

Reply:
Yeah, my test happens to use an absolute URL so it worked fine. 👍 Good call.

Reply:
Might be worth a test case although we're much safer now that the logic is in `_handle_url(...)` anyway (not a blocker)